energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -138,18 +138,23 @@ pub trait WeeklyRewardsSplittingModule:
         &self,
         week: Week,
         energy_amount: &BigUint,
-        total_rewards: &TokenAmountPairsVec<Self::Api>,
+        total_rewards: &PaymentsVec<Self::Api>,
     ) -> PaymentsVec<Self::Api> {
+        let biguint_zero = BigUint::zero();
         let mut user_rewards = ManagedVec::new();
-        if energy_amount == &0 {
+        if energy_amount == &biguint_zero {
             return user_rewards;
         }
```

Comment:
There's no need to declare the biguint_zero as a variable, comparing with `&0` works just fine. If compiler complains about comparing to `i32` (it does sometimes), then change it to `&0u32`